checks/evaluation/code_review.go

```diff
@@ -39,81 +38,57 @@ func CodeReview(name string, dl checker.DetailLogger, r *checker.CodeReviewData)
 	nUnreviewedHumanChanges := 0
-	nUnreviewedBotChanges := 0
+	nHumanChangesTotal := 0
+	unreviewedBotChanges := false
 
-	var unreviewedHumanRevisionIDs, unreviewedBotRevisionIDs []string
 	for i := range r.DefaultBranchChangesets {
 		cs := &r.DefaultBranchChangesets[i]
-		isReviewed := reviewScoreForChangeset(cs) >= changesReviewed
+		isReviewed := reviewScoreForChangeset(cs, dl) >= changesReviewed
 		isBotCommit := cs.Author.IsBot
+		if !isBotCommit {
+			nHumanChangesTotal += 1
+		}
 
 		switch {
-		case isReviewed && isBotCommit:
-			foundBotReviewActivity = true
-		case isReviewed && !isBotCommit:
-			foundHumanReviewActivity = true
 		case !isReviewed && isBotCommit:
-			nUnreviewedBotChanges += 1
-			unreviewedBotRevisionIDs = append(unreviewedBotRevisionIDs, cs.RevisionID)
+			unreviewedBotChanges = true
 		case !isReviewed && !isBotCommit:
 			nUnreviewedHumanChanges += 1
-			unreviewedHumanRevisionIDs = append(unreviewedHumanRevisionIDs, cs.RevisionID)
 		}
 	}
 
-	// Let's include non-compliant revision IDs in details
-	if len(unreviewedHumanRevisionIDs) > 0 {
-		dl.Debug(&checker.LogMessage{
-			Text: fmt.Sprintf("List of revision IDs not reviewed by humans:%s", strings.Join(unreviewedHumanRevisionIDs, ",")),
-		})
-	}
-
-	if len(unreviewedBotRevisionIDs) > 0 {
-		dl.Debug(&checker.LogMessage{
-			Text: fmt.Sprintf("List of bot revision IDs not reviewed by humans:%s", strings.Join(unreviewedBotRevisionIDs, ",")),
-		})
-	}
-
-	if foundBotReviewActivity && !foundHumanReviewActivity {
-		reason := fmt.Sprintf("found no human review activity in the last %v changesets", len(r.DefaultBranchChangesets))
-		return checker.CreateInconclusiveResult(name, reason)
-	}
-
 	switch {
-	case nUnreviewedBotChanges > 0 && nUnreviewedHumanChanges > 0:
-		return checker.CreateMinScoreResult(
-			name,
-			fmt.Sprintf("found unreviewed changesets (%v human %v bot)", nUnreviewedHumanChanges, nUnreviewedBotChanges),
-		)
-	case nUnreviewedBotChanges > 0:
+	case nHumanChangesTotal == 0:
+		reason := fmt.Sprintf("found no human review activity in the last %v changesets", N)
+		return checker.CreateInconclusiveResult(name, reason)
+	case nUnreviewedHumanChanges == 0 && unreviewedBotChanges:
 		return checker.CreateResultWithScore(
 			name,
-			fmt.Sprintf("all human changesets reviewed, found %v unreviewed bot changesets", nUnreviewedBotChanges),
+			"all human changesets reviewed, found unreviewed bot changesets",
 			7,
 		)
 	case nUnreviewedHumanChanges > 0:
-		score := 3
-		if len(r.DefaultBranchChangesets) == 1 || nUnreviewedHumanChanges > 1 {
-			score = 0
-		}
-		return checker.CreateResultWithScore(
+		return checker.CreateProportionalScoreResult(
 			name,
-			fmt.Sprintf("found %v unreviewed human changesets", nUnreviewedHumanChanges),
-			score,
+			fmt.Sprintf(
+				"found %d unreviewed changesets out of %d", nUnreviewedHumanChanges,
+				nHumanChangesTotal,
+			),
+			nHumanChangesTotal-nUnreviewedHumanChanges,
+			nHumanChangesTotal,
```

Comment:
Just to be clear, is it intentional is to ignore all unreviewed bot PRs in this score?

It leads to scenarios where having an unreviewed human PR could increase your score.
Imagine 30 PRs:
29 reviewed human PRs, and 1 unreviewed bot PRs. This would score a 7

If instead we have 28 reviewed human, 1 unreviewed human and 1 unreviewed bot, the score would now be a 28/29, which I think would be a 9.

Reply:
Yes, it's intentional to ignore all unreviewed bot PRs. Changed this so that the -3 penalty is no longer applied.